nifi-commons/nifi-properties/src/main/java/org/apache/nifi/util/NiFiProperties.java

```diff
@@ -743,6 +743,9 @@ public InetSocketAddress getClusterNodeProtocolAddress() {
                 socketAddress = "localhost";
             }
             int socketPort = getClusterNodeProtocolPort();
+            if (socketPort == 0) {
+                throw new RuntimeException("Load balance port cannot be 0. Port must be inclusively in the range [1, 65535].");
```

Comment:
The exception message indicates a problem with the Load balancing port. However, this is not checking the load balancing port - it's checking the Cluster Node Protocol Port (port used to communicate between nodes in the cluster). It does appear, though, that the exact same condition exists below, in the `getClusterLoadBalanceAddress` method.

It is possible, though, that restricting that may cause unit test or integration test failures. Often times, a port of `0` is used for such tests, so that the test is more portable, since there's no way to know if some specific port will be open in some other environment. Do we have a reason for explicitly disallowing it?

Reply:
Using port 0 throws an exception in the NodeIdentifier.java:204. Should we allow a port of 0 (a Java-chosen random port) for all port configurations in NiFi or require administrators to configure explicit ports?

Reply:
@markap14 According to NodeIdentifier.java:204, the method checks if the port is within range 1-65535. But It seems Java will handle a port 0. So, if there isn't another reason to disallow it, are we ok to change port configurations to allow port 0?

Reply:
Ahh, you've got a good point :) We cannot allow the cluster protocol port or the web (http/https) port to be set to `0` because those needs to be consistent across restarts of the application. For site-to-site port and load balancing port, these are obtained by communicating over the cluster protocol/web protocol. So if it makes things easier to just say you can't use port of 0, then that's fine. If anyone ever actually needs to, it can be revisited then. Just didn't want to restrict it if there was no reason. "It's a non-trivial amount of work" or "It's out of scope for this Jira" are perfectly valid reasons :)

Reply:
@markap14 Great. Will correct the Exception message.